### PR TITLE
 Added keycard pairing and initialization step

### PR DESCRIPTION
**Pull request by DmitryBespalov**
Sorry for the big PR.

- I moved the keycard-related submodules to MultisigWallet. After merging this and checking out - please see if you need to remove the old directories as `git` might not remove them automatically (I think).

Recommended sequence to review (in order of meat):
1. KeycardHardwareService.swift
2. KeycardApplicationService.swift
3. SKPairViewController.swift
4. SKActivateViewController.swift
5. SKPairingSuccessViewController.swift
6. SKKeycardFlowCoordinator.swift
7. DBKeycardRepository.swift

And everything else.

### CODE REVIEW

File: MultisigWallet/MultisigWalletImplementations/Services/KeycardHardwareService.swift [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+            do {
+                let cmdSet = KeycardCommandSet(cardChannel: channel)
+
+                let info = try prepare(cmdSet)
+                assert(!info.instanceUID.isEmpty, "Instance UID is not known in initialized card")
+
+                var existingPairing = DomainRegistry.keycardRepository.findPairing(instanceUID: Data(info.instanceUID))
+
+                if let pairing = existingPairing {
+                    cmdSet.pairing = Pairing(pairingKey: Array(existingPairing!.key),
```
Comment (sche) [inline MultisigWallet/MultisigWalletImplementations/Services/KeycardHardwareService.swift]: pairing.key; pairing.index

File: MultisigWallet/MultisigWalletApplication/ApplicationServiceRegistry.swift [hunk 2]
```diff
@@ -55,4 +55,10 @@ public class ApplicationServiceRegistry: AbstractRegistry {
         return service(for: EventRelay.self)
     }
 
+    // TODO: set it up in all places where needed
```
Comment (sche) [inline MultisigWallet/MultisigWalletApplication/ApplicationServiceRegistry.swift]: will you remove this with unit tests?

File: SafeUIKit/SafeAppUI/MainFlow/OnboardingFlow/SetupSafeFlow/CreateSafe/CreateSafeFlowCoordinator.swift [hunk 3]
```diff
@@ -61,7 +62,15 @@ class CreateSafeFlowCoordinator: FlowCoordinator {
...
     func didSelectTwoFAOption(_ option: TwoFAOption) {
-        self.push(OnboardingIntroViewController.createCreateSafeIntro(delegate: self))
+        switch option {
+        case .statusKeycard:
+            keycardFlowCoordinator.mainFlowCoordinator = mainFlowCoordinator
+            enter(flow: keycardFlowCoordinator) {
+                self.showSeed()
+            }
+        case .gnosisAuthenticator:
+            self.push(OnboardingIntroViewController.createCreateSafeIntro(delegate: self))
```
Comment (sche) [inline SafeUIKit/SafeAppUI/MainFlow/OnboardingFlow/SetupSafeFlow/CreateSafe/CreateSafeFlowCoordinator.swift]: this should come from my PR

File: SafeUIKit/SafeAppUI/MainFlow/OnboardingFlow/SetupSafeFlow/CreateSafe/ThreeSteps/PairWith2FAController.swift [hunk 4]
```diff
@@ -18,7 +18,7 @@ class PairWith2FAController: HeaderImageTextStepController {
     override func viewDidLoad() {
         super.viewDidLoad()
 
-        title = LocalizedString("create_safe_title", comment: "Create Safe")
+        navigationItem.title = LocalizedString("create_safe_title", comment: "Create Safe")
```
Comment (sche) [inline SafeUIKit/SafeAppUI/MainFlow/OnboardingFlow/SetupSafeFlow/CreateSafe/ThreeSteps/PairWith2FAController.swift]: what was the reason for this change?

Comment (DmitryBespalov) [reply SafeUIKit/SafeAppUI/MainFlow/OnboardingFlow/SetupSafeFlow/CreateSafe/ThreeSteps/PairWith2FAController.swift]: misunderstanding that the dark mode's white title is caused by using the title. will revert.

File: SafeUIKit/SafeAppUI/MainFlow/OnboardingFlow/SetupSafeFlow/CreateSafe/ThreeSteps/StatusKeycard/SKActivateViewController.swift [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+            self.setCredentialsLabelSelected(true)
+        }, onDisappear: { [unowned self] in
+            self.setCredentialsLabelSelected(false)
+        })
+        tooltipSource.message = LocalizedString("copied_to_clipboard", comment: "Copied to clipboard")
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        trackEvent(OnboardingTrackingEvent.pairActivation)
```
Comment (sche) [inline SafeUIKit/SafeAppUI/MainFlow/OnboardingFlow/SetupSafeFlow/CreateSafe/ThreeSteps/StatusKeycard/SKActivateViewController.swift]: please use CreateSafeTrackingEvent

File: SafeUIKit/SafeAppUI/MainFlow/OnboardingFlow/SetupSafeFlow/CreateSafe/ThreeSteps/StatusKeycard/SKPairSuccessView.swift [hunk 6]
```diff
@@ -0,0 +1,45 @@
+//
+//  Copyright © 2019 Gnosis Ltd. All rights reserved.
+//
+
+import UIKit
+import SafeUIKit
+
+class SKPairSuccessView: BaseCustomView {
```
Comment (sche) [inline SafeUIKit/SafeAppUI/MainFlow/OnboardingFlow/SetupSafeFlow/CreateSafe/ThreeSteps/StatusKeycard/SKPairSuccessView.swift]: there is already HeaderImageTextView and HeaderImageTextStepController ;)

File: SafeUIKit/SafeAppUI/MainFlow/OnboardingFlow/SetupSafeFlow/CreateSafe/ThreeSteps/StatusKeycard/SKPairingSuccessViewController.swift [hunk 7]
```diff
@@ -0,0 +1,61 @@
+//
+//  Copyright © 2019 Gnosis Ltd. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import MultisigWalletApplication
+
+class SKPairingSuccessViewController: NewSafeThreeStepsBaseController {
```
Comment (sche) [inline SafeUIKit/SafeAppUI/MainFlow/OnboardingFlow/SetupSafeFlow/CreateSafe/ThreeSteps/StatusKeycard/SKPairingSuccessViewController.swift]: You can use HeaderImageTextStepController

File: SafeUIKit/SafeAppUI/MainFlow/OnboardingFlow/Welcome/OnboardingTrackingEvent.swift [hunk 8]
```diff
@@ -48,6 +48,9 @@ enum OnboardingTrackingEvent: String, ScreenTrackingEvent {
     case twoFAScan                  = "Onboarding_2FAScan"
     case twoFAScanSuccess           = "Onboarding_2FAScanSuccess"
     case twoFAScanError             = "Onboarding_2FAScanError"
+    case pair2FADevice              = "Onboarding_Pair2FADevice"
```
Comment (sche) [inline SafeUIKit/SafeAppUI/MainFlow/OnboardingFlow/Welcome/OnboardingTrackingEvent.swift]: Please use TwoFATrackingEvent

File: MultisigWallet/MultisigWalletApplication/ApplicationServiceRegistry.swift (hunk 2 above)
Comment (DmitryBespalov) [reply MultisigWallet/MultisigWalletApplication/ApplicationServiceRegistry.swift]: yes

**Pull request merged**